src/backend/distributed/utils/reference_table_utils.c

```diff
@@ -426,62 +427,86 @@ CreateReferenceTableColocationId()
 
 
 /*
- * DeleteAllReferenceTablePlacementsFromNodeGroup function iterates over list of reference
- * tables and deletes all reference table placements from pg_dist_placement table
- * for given group.
+ * DeleteAllReplicatedTablePlacementsFromNodeGroup function iterates over
+ * list of reference and replicated hash distributed tables and deletes
+ * all placements from pg_dist_placement table for given group.
  */
 void
-DeleteAllReferenceTablePlacementsFromNodeGroup(int32 groupId)
+DeleteAllReplicatedTablePlacementsFromNodeGroup(int32 groupId, bool forceRemoteDelete)
 {
 	List *referenceTableList = CitusTableTypeIdList(REFERENCE_TABLE);
-	List *referenceShardIntervalList = NIL;
+	List *replicatedHashDistributedTableList = ReplicatedHashDistributedTableList();
```

Comment:
Hash -> MetadataSynced?

We should avoid codifying the decision on what gets synced in many different places. We're in a reasonable spot to sync append and maybe range tables.

Reply:
good point, I'm adding `ShouldSyncTableMetadata` as a check here and iterating on all the distributed tables.